bpf/bpf_lxc.c

```diff
@@ -165,6 +165,12 @@ static __always_inline int ipv6_l3_from_lxc(struct __ctx_buff *ctx,
 			*dstID = info->sec_label;
 			tunnel_endpoint = info->tunnel_endpoint;
 			encrypt_key = get_min_encrypt_key(info->key);
+#ifdef ENABLE_WIREGUARD
+			if (info->tunnel_endpoint != 0 &&
+			    info->sec_label != HOST_ID &&
+			    info->sec_label != REMOTE_NODE_ID)
+				ctx->mark |= MARK_MAGIC_ENCRYPT;
```

Comment:
Performing this before policy means that if we apply L7 egress policy, we're potentially setting the mark with all of the bits of both proxy redirect and encryption redirect.

Reply:
^ Maybe short answer is "no L7 with wireguard" and we can decide whether to be strict about this in CLI parsing or just write it up as a limitation as part of wireguard beta with undefined behaviour.

Reply:
My initial idea for setting the encrypt mark and passing to the stack instead of doing a redirect to the `cilium_wg0` was that (maybe) we could be compliant with L7. However, after taking a closer look I think I'm lacking an understanding of a packet life in our L7 world. For now, I'm going to add a check to make sure that Wireguard and L7 are mutually exclusive, and add the L7 support to the follow-up list.

Reply:
Another high-level thought about supporting the L7 proxy. Maybe we could make https://github.com/cilium/cilium/pull/11279 work without `-m socket`, so that it the mark usage is eliminated from the proxy. If the proxy could retain the mark set by bpf_lxc and would pass the packet to the stack after it has validated it against L7 policies, we could make Wireguard easily to work with the proxy. Both, BPF TPROXY and Wireguard with KPR require >= 5.7 kernel.

Reply:
> Maybe we could make #11279 work without -m socket, so that the mark usage is eliminated from the proxy path

We still need the mark to force the routing subsystem to route the proxy traffic to the local stack. Otherwise the destination address can be anything and the stack will just route the traffic to the destination instead of passing it to the proxy.

> If the proxy could retain the mark set by bpf_lxc and would pass the packet to the stack after it has validated it against L7 policies

Today, all packets from the proxy are marked with either `MARK_MAGIC_PROXY_INGRESS` or `MARK_MAGIC_PROXY_EGRESS` plus the source identity of the packet (see `inherit_identity_from_host()`). If you're suggesting that the proxy gets involved in setting the mark to forward to the encryption stack somehow, I think we're gonna have to think very carefully about this.

Reply:
> If you're suggesting that the proxy gets involved in setting the mark to forward to the encryption stack somehow

Yep, that was my thinking. OK, I think after this PR has been merged, let's have a discussion offline how we can make it work. I assume that IPSec has the same problem then?